src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelGenerator.cs

```diff
@@ -68,76 +66,86 @@ private string Generate(INamedTypeSymbol vm)
 			.ToList();
 		mappedMembers = mappedMembers.Except(mappedMembersConflictingWithInputs).ToList();
 
-		var code = $@"#nullable enable
-#pragma warning disable
 
-using System;
-using System.Linq;
-using System.Threading.Tasks;
-
-namespace {vm.ContainingNamespace}
-{{
-	partial {(vm.IsRecord ? "record" : "class")} {vm.Name} : global::System.IAsyncDisposable, {NS.Core}.ISourceContextAware
-	{{
-		public partial class Bindable{vm.Name} : {NS.Bindings}.BindableViewModelBase
-		{{
-			{inputsErrors.Align(3)}
-			{inputs.Select(input => input.GetBackingField()).Align(3)}
-			{mappedMembers.Select(member => member.GetBackingField()).Align(3)}
-
-			{vm
-				.Constructors
-				.Where(_ctx.IsGenerationNotDisable)
-				.Select(ctor =>
-				{
-					var parameters = ctor
-						.Parameters
-						.Select(parameter => inputs.First(input => input.Parameter.Name.Equals(parameter.Name, StringComparison.OrdinalIgnoreCase)))
-						.ToArray();
-
-					var bindableVmParameters = parameters
-						.Select(param => param.GetCtorParameter())
-						.Where(param => param.code is not null)
-						.OrderBy(param => param.isOptional ? 1 : 0)
-						.Select(param => param.code)
-						.JoinBy(", ");
-					var vmParameters = parameters
-						.Select(param => param.GetVMCtorParameter())
-						.JoinBy(", ");
-
-					return $@"
-						{GetCtorAccessibility(ctor)} Bindable{vm.Name}({bindableVmParameters})
-						{{
-							{inputs.Select(input => input.GetCtorInit(parameters.Contains(input))).Align(7)}
-
-							var {N.Ctor.Model} = new {vm}({vmParameters});
-							var {N.Ctor.Ctx} = {NS.Core}.SourceContext.GetOrCreate({N.Ctor.Model});
-							{NS.Core}.SourceContext.Set(this, {N.Ctor.Ctx});
-							base.RegisterDisposable({N.Ctor.Model});
-
-							{N.Model} = {N.Ctor.Model};
-							{inputs.Select(input => input.GetPropertyInit()).Align(7)}
-							{mappedMembers.Select(member => member.GetInitialization()).Align(7)}
-						}}";
-				})
-				.Align(3)}
-
-			public {vm} {N.Model} {{ get; }}
-
-			{inputs.Select(input => input.Property?.ToString()).Align(3)}
-
-			{mappedMembersErrors.Align(3)}
-			{mappedMembers.Select(member => member.GetDeclaration()).Align(3)}
-		}}
-
-		/// <inheritdoc />
-		public global::System.Threading.Tasks.ValueTask DisposeAsync()
-			=> {NS.Core}.SourceContext.Find(this)?.DisposeAsync() ?? default;
-	}}
-}}
-";
-
-		return code;
+		var bindableVmCode = $@"
+			public partial class Bindable{vm.Name} : {NS.Bindings}.BindableViewModelBase
+			{{
+				{inputsErrors.Align(4)}
+				{inputs.Select(input => input.GetBackingField()).Align(4)}
+				{mappedMembers.Select(member => member.GetBackingField()).Align(4)}
+
+				{vm
```

Comment:
could maybe extract this inline block into a variable or method, to help reduce the nesting?

Reply:
Well is not necessarily the most annoying part here ... I tried to do like in a T4 file where you generate the code inline, higher nesting, but you can almost read the generated code just by reading the file. I would say that there is no real good pattern and it's more a personal choice in that case.